### ISSUE TITLE
Introduce /search?extent.strict parameter to enable blending local + global gazetteer

### ISSUE THREAD

**Issue by riordan** [pelias/api#190]
# tl;dr
magic search. Use the same `extent` parameters often used to restrict a search to prioritize an area, while still searching the global gazetteer.

# The long version

`extent.strict` should be a new default parameter for `/search', that when disabled, will add global results in with local results from other extent scoping mechanisms (bounding box of the users screen and/or lat/lon).

This is principally to enable end-users to use the search query to control the scoping of their search, even if their device & client are severely geobiasing results.

Parameter is a flag. It is on by default. Things get more interesting when it's off.

When enabled in the presence of extent[bbox] or extent[lat/lon], the entire scope of the search is limited to the bounding box or to a radius (default 50km, but over-ridable).

When disabled, an API user is telling Pelias to infer information from the provided extent information to better scope the search for an end-user (i.e. mapzen/erasermap).

The extent parameters should be prioritized at first, but should also use query mixing to introduce major (popular) administrative areas (countries, major localities) plus global landmarks.

1. a non-strict search begins by restricting results to the general area of the bias [@riordan: DEFINE THIS FURTHER FOR WHAT IT MEANS GIVEN ZOOM LEVEL / DENSITY]. Mixed into those results are matching administrative areas

2. If a sufficient country, province, or locality is matched (e.g. it is of very high provenance or enough context is provided to decide on just one particular place), the search extent focuses to that matching place, bringing together queries from that matching area along with the device extent area.

3. Address parsing system is used to determine level/extent of match (for instance, just city vs city + state). Suggestions of places may be used to help focus this and improve extent of match

For example:
1. While in Chicago, IL, I take out my phone and fire up EraserMap. Its default state (when not currently in memory) is to zoom to show most of the city I am in. My device will pass back a bounding box of several neighborhoods around me + my current lat/lon
2. I begin to type "Brooklyn". Pelias returns as its search results "Brooklyn, Kings County, New York, USA" as well as the local ["Brooklyn Industries" store in Chicago](http://www.openstreetmap.org/node/3651400514).
3. As I continue typing __Brooklyn Nav...__, it has focused in on *Brooklyn, NY* as a principal search area, overweighting the local Chicago results (though if any match well, it will continue to prioritize those Chicago results). ["Brooklyn Navy Yard"](http://www.openstreetmap.org/way/109442847) and ["Navy Street, Brooklyn, New York..."](http://www.openstreetmap.org/way/5676939) are among top results.

The user has been able to use the query string to sufficiently see results nearby them, as well as to change the core focus of the search area.

This query mixing logic may borrow much of the code previously used for `/suggest`. See pelias/api#165 for query mixing.

This logic should also apply for `/suggest`.
